distributed/shuffle/_worker_extension.py

```diff
@@ -362,9 +359,17 @@ def _() -> dict[str, list[tuple[ArrayRechunkShardID, bytes]]]:
             write the serialized payload directly to disk on the receiver.
             """
             out: dict[str, list[tuple[ArrayRechunkShardID, bytes]]] = defaultdict(list)
-            for id, nslice in self._slicing[input_partition]:
-                out[self.worker_for[id.chunk_index]].append(
-                    (id, pickle.dumps((id.shard_index, data[nslice])))
+            from itertools import product
+
+            shards = product(
```

Comment:
nit: I believe you are using _splits_ and _shards_ interchangeably. If so, I'd prefer sticking to one

Reply:
Good catch, I've only introduced splits very recently to the chunks/chunks issue, I may have missed some cases.

Reply:
Does `ndsplits` help convey the concept that they're not truly 1-d splits nor actual shards of data?